pkg/kubectl/rolling_updater.go

```diff
@@ -175,6 +207,21 @@ func UpdateExistingReplicationController(c client.Interface, oldRc *api.Replicat
 	}
 }
 
+// scalerScaleAndWait returns a scaleAndWait function which scales a
+// controller using a Scaler and a real client.
+func scalerScaleAndWait(client RollingUpdaterClient, namespace string) scaleAndWait {
+	scaler, err := ScalerFor("ReplicationController", client)
+	return func(rc *api.ReplicationController, retry *RetryParams, wait *RetryParams) (*api.ReplicationController, error) {
+		if err != nil {
+			return nil, fmt.Errorf("Couldn't make scaler: %s", err)
+		}
+		if err := scaler.Scale(rc.Namespace, rc.Name, uint(rc.Spec.Replicas), &ScalePrecondition{-1, ""}, retry, wait); err != nil {
```

Comment:
You dont need to specify any precondition here, the scaler will initialize it for you. I was wondering though if `wait` should be specified somewhere before you return the closure (I get it that we should wait for the scaling to finish).